transformer_engine/jax/csrc/modules.cpp

```diff
@@ -91,9 +92,11 @@ pybind11::bytes PackCustomCallFusedAttnDescriptor(
 
 bool IsFusedAttnKernelAvailable() {
 #if (CUDNN_VERSION >= 8901)
-    auto major = cudaDevicePropertiesManager::Instance().GetMajor();
-    // Fused attention requires at least Ampere
-    return major >= 8;
+    int device_id = -1;
+    NVTE_CHECK_CUDA(cudaGetDevice(&device_id));
+    NVTE_CHECK(device_id >= 0, "device_id should >= 0");
+    const int sm_arch = cuda::sm_arch(device_id);
```

Comment:
This function gets the current device by default:
```suggestion
    const int sm_arch = cuda::sm_arch();
```